### PR TITLE
Added BMS messages and regen braking

### PR DESCRIPTION
**Pull request by rydzynskip**
Added BMS messages and regenerative braking.

### CODE REVIEW

**Review by mccauleyma — COMMENTED** [review]
Initial impression is good. I just need to know what library to install so I can give it a compile on my machine.

**Review by mccauleyma — APPROVED** [review]
Looks great. Pending testing with the actual CAN installation

File: ArduinoCode/Pedal_Potentiometers/Pedal_Potentiometers.ino [hunk 1]
```diff
@@ -95,50 +121,52 @@ void readPotentiometers() {
...
-        invertorOn = true;
-      }
-    } else if (canId == 0x02) {  // message about forward/reverse
-      if (buf[0] == 0) {
-        isForward = false;
-      } else if (buf[0] == 1) {
-        isForward = true;
-      }
+
+    if (canId == CAN_POWER) {
```
Comment (jamescd18) [inline ArduinoCode/Pedal_Potentiometers/Pedal_Potentiometers.ino:152]: what does CAN_POWER mean?

**Review by jamescd18 — CHANGES_REQUESTED** [review]
Overall pretty good! There's a few things I'm curious to get clarification on, but it's minor. Also, not necessarily saying you should do this, but it's easier to read if you order your code in a consistent manner across the function declarations, usages, and function definitions. Basically, since the usage is as simple as looking at the loop() function to see readStates(), readPotentiometers(), and readSwitches(), I would personally choose to use that order in the function declarations and in the function definitions. Again, not suggesting you need to or anything, just point it out.

File: ArduinoCode/Pedal_Potentiometers/Pedal_Potentiometers.ino [hunk 2]
```diff
@@ -95,50 +121,52 @@ void readPotentiometers() {
...
+ * readStates() reads the CAN messages from other parts of the car to set the following states:
+ *  - CAN_POWER message is used to set the inverterOn state
+ *  - CAN_DIRECTION message is used to set the isForward state
+ *  - CAN_BMS_STATES message is used to set the dischargeEnabled and chargeEnabled states
  */
-void readDashboard() {
+void readStates() {
   if(CAN_MSGAVAIL == CAN.checkReceive()) {   // if a new message has been recieved.
-    CAN.readMsgBuf(&len, buf); // enters message into program
     canId = CAN.getCanId(); // gets sender ID
```
[the remark this answers is not in the document]
Comment (rydzynskip) [reply ArduinoCode/Pedal_Potentiometers/Pedal_Potentiometers.ino:150]: No you can't, good catch. I'll change this

**Review by jamescd18 — APPROVED** [review]
Perfect! Makes much more sense now. Also, yeah, saw the buf[8] vs buf[1], but I didn't know enough about C++ to be sure that it could be simplified to just buf[8] which is why I didn't mention it, so great catch there. Looks very good overall, and just next time you can resolve conversation points / comment threads if you feel like you've fixed it or are done with it

**Pull request merged**